packages/manager/src/components/RegionSelect/TwoStepRegionSelect.tsx

```diff
@@ -0,0 +1,139 @@
+import { TabList } from 'src/components/Tabs/TabList';
+import { TabPanels } from 'src/components/Tabs/TabPanels';
+import { Tabs } from 'src/components/Tabs/Tabs';
+import { sendLinodeCreateDocsEvent } from 'src/utilities/analytics/customEventAnalytics';
+
+import type {
+  DisableRegionOption,
+  RegionFilterValue,
+} from './RegionSelect.types';
+import type { Region } from '@linode/api-v4';
+import type { SelectRegionPanelProps } from 'src/components/SelectRegionPanel/SelectRegionPanel';
+
+interface TwoStepRegionSelectProps
+  extends Omit<SelectRegionPanelProps, 'selectedId'> {
+  disabledRegions: Record<string, DisableRegionOption>;
+  regions: Region[];
+  value?: string;
+}
+
+interface GeographicalAreaOption {
+  label: string;
+  value: RegionFilterValue;
+}
+
+const GEOGRAPHICAL_AREA_OPTIONS: GeographicalAreaOption[] = [
+  {
+    label: 'All',
+    value: 'distributed-ALL',
+  },
+  {
+    label: 'North America',
+    value: 'distributed-NA',
+  },
+  {
+    label: 'Africa',
+    value: 'distributed-AF',
+  },
+  {
+    label: 'Asia',
+    value: 'distributed-AS',
+  },
+  {
+    label: 'Europe',
+    value: 'distributed-EU',
+  },
+  {
+    label: 'Oceania',
+    value: 'distributed-OC',
+  },
+  {
+    label: 'South America',
+    value: 'distributed-SA',
+  },
+];
+
+export const TwoStepRegionSelect = (props: TwoStepRegionSelectProps) => {
+  const {
+    RegionSelectProps,
+    currentCapability,
+    disabled,
+    disabledRegions,
+    error,
+    handleSelection,
+    helperText,
+    regions,
+    value,
+  } = props;
+
+  const [regionFilter, setRegionFilter] = React.useState<RegionFilterValue>(
+    'distributed'
+  );
+
+  return (
+    <Tabs>
+      <TabList>
+        <Tab>Core</Tab>
+        <Tab>Distributed</Tab>
+      </TabList>
+      <TabPanels>
+        <SafeTabPanel index={0}>
+          <Box marginTop={2}>
+            <RegionHelperText
+              onClick={() => sendLinodeCreateDocsEvent('Speedtest')}
+            />
+          </Box>
+          <RegionSelect
+            currentCapability={currentCapability}
+            disableClearable
+            disabled={disabled}
+            errorText={error}
+            helperText={helperText}
+            onChange={(e, region: Region) => handleSelection(region.id)}
+            regionFilter="core"
+            regions={regions ?? []}
+            showDistributedRegionIconHelperText={false}
+            disabledRegions={disabledRegions}
+            value={value}
+            {...RegionSelectProps}
+          />
+        </SafeTabPanel>
+        <SafeTabPanel index={1}>
+          <Autocomplete
+            onChange={(_, selectedOption) => {
+              if (selectedOption?.value) {
+                setRegionFilter(selectedOption.value);
+              }
+            }}
+            defaultValue={GEOGRAPHICAL_AREA_OPTIONS[0]}
+            disableClearable
+            label="Geographical Area"
+            options={GEOGRAPHICAL_AREA_OPTIONS}
```

Comment:
Should we generate the options based on the regions? 

```suggestion
            options={Array.from(
              new Set(regions.map((region) => getRegionCountryGroup(region)))
            ).map((value) => ({ label: value }))}
```

This will require other changes, but this might be worth the investment if we expect this code to be around long term.

Reply:
Hmm with how we're currently doing the region filtering in `getRegionOptions`, we would need to duplicate getting the options or refactor `RegionSelect` to accept a `regionOptions` prop. I'm leaning more toward the latter, what do you think?

Either way, I think the lift is out of scope for this ticket but I did create a ticket for refactoring the `getRegionOptions` function (M3-8292)

Reply:
I was able to accomplish this by just modifying `TwoStepRegionSelect`, but we can save this for later to keep the scope narrow if needed!

<details>
  <summary>Code</summary>
  
 Could use some improvement/clean up and better type-safety, but this is what I was thinking
```typescript
import * as React from 'react';

import { Autocomplete } from 'src/components/Autocomplete/Autocomplete';
import { Box } from 'src/components/Box';
import { RegionSelect } from 'src/components/RegionSelect/RegionSelect';
import { RegionHelperText } from 'src/components/SelectRegionPanel/RegionHelperText';
import { SafeTabPanel } from 'src/components/Tabs/SafeTabPanel';
import { Tab } from 'src/components/Tabs/Tab';
import { TabList } from 'src/components/Tabs/TabList';
import { TabPanels } from 'src/components/Tabs/TabPanels';
import { Tabs } from 'src/components/Tabs/Tabs';
import { sendLinodeCreateDocsEvent } from 'src/utilities/analytics/customEventAnalytics';
import { getRegionCountryGroup } from 'src/utilities/formatRegion';

import type { DisableRegionOption } from './RegionSelect.types';
import type { Region } from '@linode/api-v4';
import type { SelectRegionPanelProps } from 'src/components/SelectRegionPanel/SelectRegionPanel';

interface TwoStepRegionSelectProps
  extends Omit<SelectRegionPanelProps, 'selectedId'> {
  disabledRegions: Record<string, DisableRegionOption>;
  regions: Region[];
  value?: string;
}

export const TwoStepRegionSelect = (props: TwoStepRegionSelectProps) => {
  const {
    RegionSelectProps,
    currentCapability,
    disabled,
    disabledRegions,
    error,
    handleSelection,
    helperText,
    regions,
    value,
  } = props;

  const [
    selectedGrographicalArea,
    setGrographicalArea,
  ] = React.useState<string>();

  const geographicalAreaOptions = Array.from(
    new Set(regions.map((region) => getRegionCountryGroup(region)))
  ).map((value) => ({ label: value }));

  return (
    <Tabs>
      <TabList>
        <Tab>Core</Tab>
        <Tab>Distributed</Tab>
      </TabList>
      <TabPanels>
        <SafeTabPanel index={0}>
          <Box marginTop={2}>
            <RegionHelperText
              onClick={() => sendLinodeCreateDocsEvent('Speedtest')}
            />
          </Box>
          <RegionSelect
            currentCapability={currentCapability}
            disableClearable
            disabled={disabled}
            disabledRegions={disabledRegions}
            errorText={error}
            helperText={helperText}
            onChange={(e, region: Region) => handleSelection(region.id)}
            regionFilter="core"
            regions={regions ?? []}
            showDistributedRegionIconHelperText={false}
            value={value}
            {...RegionSelectProps}
          />
        </SafeTabPanel>
        <SafeTabPanel index={1}>
          <Autocomplete
            onChange={(_, selectedOption) => {
              setGrographicalArea(selectedOption?.label);
            }}
            label="Geographical Area"
            options={geographicalAreaOptions}
            placeholder="All"
          />
          <RegionSelect
            regions={
              selectedGrographicalArea
                ? (regions ?? []).filter(
                    (r) => getRegionCountryGroup(r) === selectedGrographicalArea
                  )
                : regions
            }
            currentCapability={currentCapability}
            disableClearable
            disabled={disabled}
            disabledRegions={disabledRegions}
            errorText={error}
            helperText={helperText}
            onChange={(e, region: Region) => handleSelection(region.id)}
            regionFilter="distributed"
            showDistributedRegionIconHelperText={false}
            value={value}
            {...RegionSelectProps}
          />
        </SafeTabPanel>
      </TabPanels>
    </Tabs>
  );
};
```

</detail>

Reply:
Ahh gotcha, this works but now there are two ways to filter regions (filtering regions before passing it in and then filtering in `getRegionOptions`. I think we should stick with one way but yeah let's save this for later!